libs/src/services/ethContracts/index.js

```diff
@@ -159,70 +159,80 @@ class EthContracts {
    * @return {Promise<string>} A valid service provider url with the fastest response
    */
   async selectLatestServiceProvider (spType, whitelist = null) {
-    let discoveryProviders = await this.ServiceProviderFactoryClient.getServiceProviderList(spType)
+    let serviceProviders = await this.ServiceProviderFactoryClient.getServiceProviderList(spType)
     if (whitelist) {
-      discoveryProviders = discoveryProviders.filter(d => whitelist.has(d.endpoint))
+      serviceProviders = serviceProviders.filter(d => whitelist.has(d.endpoint))
     }
 
     // No discovery providers found.
-    if (discoveryProviders.length === 0) {
+    if (serviceProviders.length === 0) {
       return null
     }
 
-    let selectedDiscoveryProvider
-    try {
-      selectedDiscoveryProvider = await Utils.promiseFight(
-        discoveryProviders.map(async (discprov) => {
-          try {
-            const healthResp = await axios({ url: urlJoin(discprov.endpoint, 'health_check'), method: 'get' })
-            const { status, block_difference: blockDiff } = healthResp
-            if (
-              status !== 200 ||
-              blockDiff > UNHEALTHY_BLOCK_DIFF
-            ) {
-              throw new Error(`Discprov healthcheck failed ${discprov.endpoint}`)
-            }
-
-            const {
-              data: { service: serviceName, version: serviceVersion }
-            } = await axios({ url: urlJoin(discprov.endpoint, 'version'), method: 'get' })
+    // Load expected version information if necessary
+    if (!this.expectedServiceVersions) {
+      this.expectedServiceVersions = await this.getExpectedServiceVersions()
+    }
+    if (!this.expectedServiceVersions.hasOwnProperty(spType)) {
+      throw new Error(`Invalid service name: ${spType}`)
+    }
 
-            // Compare chain service name
-            if (!this.expectedServiceVersions) {
-              this.expectedServiceVersions = await this.getExpectedServiceVersions()
-            }
-            if (!this.expectedServiceVersions.hasOwnProperty(serviceName)) {
-              throw new Error(`Invalid service name: ${serviceName}`)
-            }
+    let expectedVersion = this.expectedServiceVersions[spType]
 
-            if (serviceName !== spType) {
-              throw new Error(`Invalid service type: ${serviceName}. Expected ${spType}`)
-            }
+    let selectedServiceProvider
+    try {
+      // Querying all versions
+      let foundVersions = new Set()
+      let spVersionToEndpoint = {}
+      await Promise.all(serviceProviders.map(async (sp) => {
+        const {
```

Comment:
@dmanjunath - should this entire endpoint be try catch'd so a single timeout doesn't end other potential successful queries?

Reply:
have added it here, lmk